core/src/test/java/org/apache/iceberg/TestDeleteFiles.java

```diff
@@ -412,6 +414,34 @@ public void testDeleteWithCollision() {
         afterDeletePartitions);
   }
 
+  @Test
+  public void testDeleteValidateFileExistence() {
+    commit(table, table.newFastAppend().appendFile(FILE_B), branch);
+    Assertions.assertThatThrownBy(
```

Comment:
Why is this test using concurrent deletes?

Reply:
I'm working on adding a test which does the rewriteDataFiles + deleteFiles concurrently as well ( the scenario in the description), but I think the same idea applies for concurrent deletes. Concurrent deletes with validation enabled should still fail because one delete will succeed, and then one which has to retry will fail the validation.

Reply:
I just was wondering why you wouldn't just do them serially, shouldn't that be exactly the same? I also don't think we need a rewrite + delete test. All we really need to verify is the exception when the file isn't there, the cause isn't really important correct?

Reply:
Yes you're right.  we just care about checking the validate API behavior in these tests. So we can do one call which performs the delete , followed by another call with validation which should fail. The concurrency part is not really relevant for the purpose of tests